modules/gapi/test/streaming/gapi_gstreamersource_tests.cpp

```diff
@@ -377,7 +449,96 @@ TEST_P(GStreamerMultiSourceTest, ImageDataTest)
+                                       })),
+                               Values(cv::gapi::wip::GStreamerSource::OutputType::FRAME)));
+
+struct GStreamerMultiSourceTestGRAY8 :
+    public TestWithParam<std::tuple<cv::GComputation, cv::gapi::wip::GStreamerSource::OutputType>>
+{ };
+
+TEST_P(GStreamerMultiSourceTestGRAY8, ImageDataTest)
+{
+    std::string pathToLeftIm = findDataFile("cv/stereomatching/datasets/tsukuba/im6.png");
+    std::string pathToRightIm = findDataFile("cv/stereomatching/datasets/tsukuba/im2.png");
+
+    std::string pipelineToReadImage("filesrc location=LOC ! pngdec ! videoconvert ! "
+        "videoscale ! video/x-raw,format=GRAY8 ! appsink");
+
+    cv::gapi::wip::GStreamerSource leftImageProvider(
+        std::regex_replace(pipelineToReadImage, std::regex("LOC"), pathToLeftIm));
+    cv::gapi::wip::GStreamerSource rightImageProvider(
+        std::regex_replace(pipelineToReadImage, std::regex("LOC"), pathToRightIm));
+
+    cv::gapi::wip::Data leftImData, rightImData;
+    leftImageProvider.pull(leftImData);
+    rightImageProvider.pull(rightImData);
+
+    cv::Mat leftRefMat =  cv::util::get<cv::Mat>(leftImData);
+    cv::Mat rightRefMat = cv::util::get<cv::Mat>(rightImData);
+
+    // Retrieve test parameters:
+    std::tuple<cv::GComputation, cv::gapi::wip::GStreamerSource::OutputType> params = GetParam();
+    cv::GComputation extractImage = std::move(std::get<0>(params));
+    cv::gapi::wip::GStreamerSource::OutputType outputType = std::get<1>(params);
+
+    // Graph compilation for streaming mode:
+    auto compiled =
+        extractImage.compileStreaming();
+
+    EXPECT_TRUE(compiled);
+    EXPECT_FALSE(compiled.running());
+
+    cv::gapi::wip::GStreamerPipeline
+        pipeline(std::string("multifilesrc location=" + pathToLeftIm + " index=0 loop=true ! "
+                 "pngdec ! videoconvert ! videoscale ! video/x-raw,format=GRAY8 ! "
+                 "appsink name=sink1 ") +
+                 std::string("multifilesrc location=" + pathToRightIm + " index=0 loop=true ! "
+                 "pngdec ! videoconvert ! videoscale ! video/x-raw,format=GRAY8 ! "
+                 "appsink name=sink2"));
+
+    // GStreamer streaming sources configuration:
+    auto src1 = pipeline.getStreamingSource("sink1", outputType);
+    auto src2 = pipeline.getStreamingSource("sink2", outputType);
+
+    compiled.setSource(cv::gin(src1, src2));
+
+    // Start of streaming:
+    compiled.start();
+    EXPECT_TRUE(compiled.running());
+
+    // Streaming - pulling of frames:
+    cv::Mat in_mat1, in_mat2;
+
+    std::size_t counter { }, limit { 10 };
+    while(compiled.pull(cv::gout(in_mat1, in_mat2)) && (counter < limit)) {
+        EXPECT_EQ(0, cv::norm(in_mat1, leftRefMat, cv::NORM_INF));
+        EXPECT_EQ(0, cv::norm(in_mat2, rightRefMat, cv::NORM_INF));
+        ++counter;
+    }
+
+    compiled.stop();
+
+    EXPECT_FALSE(compiled.running());
+}
```

Comment:
So the test is pretty identical to NV12 one - can they be merged into one, with a flag `isNv12` as previous ones?

Or, as it is a test, it doesn't really worth it to spend time on that?..

Reply:
Considered it but can't find easy way to do so. So just want to overcomplicate test code and make it potentially error prone.  Moreover, I 'don't really like my approach with isNv12 . Seems it's not really scalable.